### ISSUE TITLE
Impossible to initialize non-packed empty structs.

### ISSUE THREAD

**Issue by lolbinarycat** [ziglang/zig#7500]
as far as I know, there is no way to construct a value with type `struct {}`. normal struct initialization fails:
```
./src/main.zig:6:18: error: expected type 'struct:6:9', found 'void'
    _ = struct {}{};
                 ^
```

as does any type of coersion.

**Comment by ehaas** [comment]
It does seem to work if you assign to something other than `_` (tested on 84549b426 on macOS Big Sur)
```zig
pub fn main() void {
    const foo = struct{}{};
    @import("std").debug.print("{}\n", .{foo});
}
```
```
➜  ~ zig build-exe test.zig
➜  ~ ./test
foo{ }
```

**Comment by mikdusan** [comment]
- latest compiler no longer emits error
- seems it was fixed somewhere between 0.9.0 and 0.10.0

<details>
<summary>repro.zig</summary>

```zig
export fn foo() void {
    _ = struct {}{};
}
```
</details>

```sh
$ uname -a
Linux archlinux 6.1.2-arch1-1 #1 SMP PREEMPT_DYNAMIC Sat, 31 Dec 2022 17:40:35 +0000 x86_64 GNU/Linux

$ zig version
0.11.0-dev.1239+7a2d7ff62

$ zig build-obj repro.zig
# no stdout/stderr (good)
```
